### PR TITLE
Rename "text input" to "input field"

### PR DESCRIPTION
**Pull request by hannalaakso**
We've published some [guidance about how to ask users for numbers](https://github.com/alphagov/govuk-design-system/pull/1144). 

This updates the component name in the guidance as discussed in https://github.com/alphagov/govuk-design-system/issues/1129. 

This PR:
- Renames "text input" to "input field"
- Adds the aliases "text input" and "number input" for input field component
- Adds a redirect for `/components/text-input`

Fixes https://github.com/alphagov/govuk-design-system/issues/1129

### CODE REVIEW

File: src/components/input-field/index.md.njk [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
-title: Text input
-description: Help users enter information with the text input component
+title: Input field
+description: Help users enter information with the input field component
 section: Components
 aliases: text box, text field, input field, text entry box
```
Comment (nickcolley) [inline src/components/input-field/index.md.njk]: could we change the 'input field' alias to 'text input' now?

Comment (hannalaakso) [reply src/components/input-field/index.md.njk]: Good catch. Do you think we should add 'number input' as an alias now too? 🤔

Comment (nickcolley) [reply src/components/input-field/index.md.njk]: Maybe, I'll leave that for you to decide 🤔

Comment (hannalaakso) [reply src/components/input-field/index.md.njk]: I've gone with adding it. This will go through a content review too so we'll look at it there.

**Comment by StephenGill** [comment]
Naming things is hard.

I'd suggest that ideally, our name for a pattern, style or component would be:

- consistent with the way our users talk and think about the thing (so it's easier to find and understand)
- easy to differentiate from other Design System patterns, styles and components
- avoids confusion with other naming conventions (for example, the HTML specification)
- is consistent with GOV.UK Frontend

Suggestions for this component were:

- leave it as it is (ie 'Text input')
- change it to 'Text or number input'
- change it to 'Input field'

I think 'Input field' is the best option because it conveys the sense of a blank space where you enter text or a number.

It's not perfect (for example, it moves us away from the terminology used in the HTML spec a little). But I don't think there's a perfect solution in this case.

Happy to talk about it if people have different views, though.

**Comment by 36degrees** [comment]
So, I originally suggested that we consider making a change for two reasons:

1) There's a discrepancy between what we call the component in the Design System ("Text Input") and what we call it in Frontend (govukInput). This is the only component that has that discrepancy.
2) It felt like, by adding guidance around asking for numbers, it _might_ be confusing for users who wouldn't expect to find guidance about numbers in a component called 'Text Input'. (Perhaps even more more confusingly, it's also the component that you'd use for a password input, although we don't provide any guidance or examples that show this)

These are both based on some fairly big assumptions around what users may or may not find confusing, without any real evidence.

Especially given that renaming the component to 'Input field' does not address point 1, my personal preference would be to leave it as 'Text Input' until we are able to do further research or otherwise get some more insight into this. To be honest, at this point I slightly regret ever raising it 😆. However, that's just my personal preference and if we think there's good reasons for making the change, I'm happy to go with it.

**Comment by joelanman** [comment]
Agree there's no perfect solution, but personally I think there's value in trying to stay close to the html spec (personally I think we should look into an alias for `govukInput` - `govukTextInput` for this reason).

Agree with Ollie that Input Field doesn't fix this discrepancy issue, so maybe stick with what we have for now?

**Comment by 36degrees** [comment]
Closing this as we've decided not to do it – will be documented in #1129.

**Pull request closed without merging**